### ISSUE TITLE
Custom parser documentation doesn't specify Read + Seek bounds come from binread::io not std::io

### ISSUE THREAD

**Issue by anieuwland** [jam1garner/binread#23]
I'm trying to define a custom parser, but I'm getting typing errors that make it impossible to do any parsing. Basically, it looks like this:

```rust
use std::io::Read;
use std::io::Seek;
use byteorder::{BigEndian, ReadBytesExt};
use binread::*;
// (others left out for brevity, removing imports doesn't solve issue however)

#[derive(Debug)]
#[derive(BinRead)]
struct FlirRawData {
    #[br(pad_before = 2)]
    #[br(little)]
    image_width: u16,
    #[br(little)]
    image_height: u16,
    image_type: u16,
    #[br(pad_before = 26)]
    #[br(parse_with = raw_image_parser)]
    raw_image Vec<u8>,
}

fn raw_thermal_parser<R: Read + Seek>(reader: &mut R, ro: &ReadOptions, _: ()) -> BinResult<Vec<u8>> {
    Ok(Vec::new())
}
```

Basically, it's like straight from the [documentation](https://docs.rs/binread/1.1.1/binread/attribute/index.html#custom-parsers). Unfortunately this fails to compile with the error that the trait bounds aren't satisfied (as below). If I remove the `Read` trait from the `R` type in the function definition, then it complains about `Seek` not being implemented for `R`. If I remove both, then it compiles, but then R can't be used for parsing in any way that I can see.

```
error[E0277]: the trait bound `R: std::io::Read` is not satisfied
   --> src/thermograms/flir.rs:174:23
    |
174 |     #[br(parse_with = raw_image_parser)]
    |                       ^^^^^^^^^^^^^^^^^^ the trait `std::io::Read` is not implemented for `R`
```

What's going wrong here?

**Comment by jam1garner** [comment]
Ah ok! I should definitely clarify that in the docs. But `Read + Seek` there refers to `binread::io::{Read, Seek};`, which are what allows `binread` to work in no_std.

If you change your imports to use those, it will work. The interface should be the same, and `binread::io::Read` is implemented for `std::io::Read`, but unfortunately since all of `std::io` is std-only this is a bit of a hack that's kinda required.

Keeping this issue open as a documentation issue though until I (or someone else) gets around to fixing it.

**Comment by Nimmerwoner** [comment]
Thanks, that works! This way, I can't figure out how to read a file into a `Vec`, however. I used to do the following:

```rust
    let mut stream = File::open(file_path)?;
    let mut bytes: Vec<u8> = Vec::with_capacity(file.metadata().unwrap().len() as usize);
    file.read_exact(&mut bytes)?;
    println!("{:?}", bytes);
```
But now `read_exact` isn't available anymore. But if I change `read_exact` to `read`, the code doesn't read anything (that is, the output of the final line is `[]`). How would you read in a binary file in combination with binread's `Read`?

**Comment by jam1garner** [comment]
Marking this as closed due to #38 making this _probably_ not needed. Reopen if it's a problem for you still.
